### PR TITLE
Core 1941/super gradients integration

### PR DESCRIPTION
**Pull request by pravin-unstructured**
Add new detection class which allows the user to supply a super-gradients model with the following values;
1. model_arch supported by super-gradients
2. model_path to the model weights
3. yaml file with a names section containing in order the labels for the classes to the object detection model
4. callback function which takes an image and a super-gradients model and returns supervision detections

Wlll be used for AF contract and contracts in the future.

### CODE REVIEW

File: requirements/base.in [hunk 1]
```diff
@@ -2,7 +2,8 @@ layoutparser[layoutmodels,tesseract]
 python-multipart
 huggingface-hub
 opencv-python!=4.7.0.68
-onnx==1.14.1
+super-gradients
```
Comment (MthwRobinson) [inline requirements/base.in:5]: Will most `unstructured` users use the super gradients functionality? If not we may want to make these extras to keep the base installation as slim as possible.

**Review by MthwRobinson — COMMENTED** [review]
Main thought from me is that we probably want to make the new dependencies extras if only a limited set of users will use the super gradients functionality.

**Review by MthwRobinson — APPROVED** [review]
LGTM! Discussed a few things offline w/ @qued

**Review by qued — CHANGES_REQUESTED** [review]
Adding a hold to make sure release happens before this merges.

**Review by qued — APPROVED** [review]

**Pull request merged**